designsystem/src/androidUnitTest/kotlin/org/hisp/dhis/mobile/ui/designsystem/InputAgeSnapshotTest.kt

```diff
@@ -22,74 +23,81 @@ class InputAgeSnapshotTest {
             ColumnScreenContainer {
                 SubTitle("Input Age Component - Idle")
                 InputAge(
-                    InputAgeModel(
-                        title = "Label",
-                        inputType = AgeInputType.None,
-
-                        onValueChanged = {
-                        },
+                    state = rememberInputAgeState(
+                        inputAgeData = InputAgeData(
+                            title = "Label",
+                        ),
                     ),
+                    onValueChanged = {
+                    },
                 )
 
                 SubTitle("Input Age Component - Idle Disabled")
                 InputAge(
-                    InputAgeModel(
-                        title = "Label",
-                        inputType = AgeInputType.None,
-                        state = InputShellState.DISABLED,
-                        onValueChanged = {
-                        },
+                    state = rememberInputAgeState(
+                        inputAgeData = InputAgeData(
+                            title = "Label",
+                        ),
+                        inputState = InputShellState.DISABLED,
                     ),
+                    onValueChanged = {
+                    },
                 )
 
                 SubTitle("Input Age Component - Date Of Birth")
                 InputAge(
-                    InputAgeModel(
-                        title = "Label",
+                    state = rememberInputAgeState(
+                        inputAgeData = InputAgeData(
+                            title = "Label",
+                        ),
                         inputType = AgeInputType.DateOfBirth(
                             TextFieldValue("01011985"),
                         ),
```

Comment:
Could you add some examples of the input receiving the expected value? the new implementation should receive a value like  "1991-11-27" format it internally to "27111991" and return the first format again whenever the value changes